tests/host_callback_test.py

```diff
@@ -103,6 +128,40 @@ def repl_func(match_group):
   tst.assertMultiLineStrippedEqual(expected, what)
 
 
+def assertMultiDeviceOutputEqual(tst: jtu.JaxTestCase,
+                                 expected_2CPUs: str):
+  """Check that the multi-device output is equal to the expected.
+
+  Args:
+    expected_2CPUs: the expected output for 2 CPUs. If there is only
+      one device, this is trimmed to the first CPU. If the current
+      device_under_test is not a CPU, then we change the names
+  """
+  assert api.device_count() in (1, 2)
+  expected = expected_2CPUs
+  if api.device_count() == 1:
+    start_device_1 = expected.find('device: cpu:1')
+    if start_device_1 >= 0:
+      expected = expected[0:start_device_1]
+
+  def replace_device_name(m) -> str:
+    return str(api.devices()[int(m.group(1))])
+  expected = re.sub(r'cpu:(\d+)', replace_device_name, expected)
+  what = testing_stream.output_sorted_by_device
+  return assertMultiLineStrippedEqual(tst, expected, what)
+
+
+# Run all tests with 8 CPU devices.
```

Comment:
is this a typo? It looks like it's always 2 CPU devices?

Reply:
A typo indeed, fixed. Thanks!